src/rust/aggregator/src/pipelines/order_history_pipelines.rs

```diff
@@ -0,0 +1,594 @@
+                            < (change.txn_version.clone(), change.event_idx.clone())
+                        {
+                            (Some(fill), None)
+                        } else {
+                            (None, Some(change))
+                        }
+                    }
+                    (None, None) => unreachable!(),
+                    other => other,
+                };
+                match (f, c) {
+                    (Some(fill), None) => {
+                        self.last_prices
+                            .insert(fill.market_id.clone(), Some(fill.price.clone()));
+                        handle_fill(fill, orders, &mut fills_index);
+                    }
+                    (None, Some(change)) => {
+                        handle_change(change, orders, &mut changes_index);
+                    }
+                    _ => unreachable!(),
+                };
+            }
+
+            *txn_version_of_state = txn_version;
+        }
+    }
+}
+
+/// Returns the min ask and max bid per market from a given state.
+fn get_spreads_from_state<'a>(
+    state: &SpreadsState,
+) -> (
+    HashMap<BigDecimal, BigDecimal>,
+    HashMap<BigDecimal, BigDecimal>,
+) {
+    let mut min_asks: HashMap<BigDecimal, BigDecimal> = Default::default();
+    let mut max_bids: HashMap<BigDecimal, BigDecimal> = Default::default();
+
+    for element in state.orders.values() {
+        if element.side == ASK {
+            if let Some(min) = min_asks.get(&element.market_id) {
+                if *min > element.price {
+                    min_asks.insert(element.market_id.clone(), element.price.clone());
+                }
+            } else {
+                min_asks.insert(element.market_id.clone(), element.price.clone());
+            }
+        } else {
+            if let Some(max) = max_bids.get(&element.market_id) {
+                if *max < element.price {
+                    max_bids.insert(element.market_id.clone(), element.price.clone());
+                }
+            } else {
+                max_bids.insert(element.market_id.clone(), element.price.clone());
+            }
+        }
+    }
+
+    (min_asks, max_bids)
+}
+
+/// Returns the amount of base and quote per group per BPS from a given state.
+///
+/// hashmap type:
+///
+/// - key: (group_id, bps)
+/// - value: (base_amount, quote_amount)
+async fn get_liquidities_from_state<'a>(
+    addr_to_gr: &HashMap<String, i32>,
+    market_id_to_gr: &HashMap<BigDecimal, i32>,
+    transaction: &Arc<RwLock<Transaction<'a, Postgres>>>,
+    state: &SpreadsState,
+    last_prices: &mut HashMap<BigDecimal, Option<BigDecimal>>,
+) -> Result<HashMap<LiquidityKey, LiquidityValue>, PipelineError> {
+    let mut liquidities: HashMap<LiquidityKey, LiquidityValue> = Default::default();
+
+    let orders = &state.orders;
+
+    if orders.is_empty() {
+        return Ok(liquidities);
+    }
+
+    let bpss = vec![0, 5, 10, 25, 50, 100, 200];
```

Comment:
As discussed on previous eng call this should include 2.5

Also, this is a magic number that is better defined as a constant

- [ ] Include 2.5 bps
- [ ] Define as constant outside this function